cuda_core/docs/source/api.rst

```diff
@@ -75,6 +75,7 @@ Utility functions
 
    args_viewable_as_strided_memory
 
-   :template: dataclass.rst
```

Comment:
* dataclass.rst does not render methods.
* class.rst omits cythonized properties

cyclass places attributes section just after the main class docstring. this way we can document the actual attributes at the end of the main docstring and they are followed by docstring of all the properties.